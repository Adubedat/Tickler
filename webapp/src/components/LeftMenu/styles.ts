import styled from 'styled-components';

export const LeftMenuContainer = styled.div`
  position: relative;
  display: flex;
  flex: 1;
  max-width: 200px;
  flex-direction: column;
  background-color: #394b59;
`;

export const MenuElement: any = styled.div`
  color: white;
  display: flex;
  justify-content: space-between;
  padding: 15px;
  :hover {
    background-color: #31414d;
  }
  background-color: ${(props: any) => {
    if (props.id === props.selected)
      return '#31414d'
  }}
`;