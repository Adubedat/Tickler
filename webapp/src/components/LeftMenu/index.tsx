import React, { useState } from 'react';
import { Icon, Intent } from "@blueprintjs/core";
import { IconNames } from "@blueprintjs/icons";
import { LeftMenuContainer, MenuElement } from './styles';
import { useUserState } from '../../context/User';

interface Props {
  propsSelected?: string;
}

const LeftMenu: React.FunctionComponent<Props> = ({ propsSelected }) => {
  const [selected, setSelected] = useState(propsSelected || 'tickets')
  const user = useUserState();

  const handleClick = (e: { target: HTMLInputElement }) => {
    setSelected(e.target.id);
  }

  const displayStatistics = () => {
    if (user.role === "admin") {
      return (
        <MenuElement id="statistics" selected={selected} onClick={handleClick}>
          <Icon icon={IconNames.CHART} iconSize={Icon.SIZE_LARGE} intent={Intent.NONE} style={{display: 'flex', alignItems: 'center'}} />
          <span style={{ fontSize: '1.1em', display: 'flex', alignItems: 'center'}}>Statistics</span>
          <Icon icon={IconNames.CHEVRON_RIGHT} iconSize={Icon.SIZE_LARGE} intent={Intent.NONE} style={{display: 'flex', alignItems: 'center'}}/>
        </MenuElement>
    );
    }
  }

  return (
    <LeftMenuContainer >
      <MenuElement id="tickets" selected={selected} onClick={handleClick}>
        <Icon icon={IconNames.BOOKMARK} iconSize={Icon.SIZE_LARGE} intent={Intent.NONE} style={{display: 'flex', alignItems: 'center'}} />
        <span style={{ fontSize: '1.1em', display: 'flex', alignItems: 'center'}}>Tickets</span>
        <Icon icon={IconNames.CHEVRON_RIGHT} iconSize={Icon.SIZE_LARGE} intent={Intent.NONE} style={{display: 'flex', alignItems: 'center'}}/>
      </MenuElement>
      {displayStatistics()}
    </LeftMenuContainer>
  );
};

export default LeftMenu;