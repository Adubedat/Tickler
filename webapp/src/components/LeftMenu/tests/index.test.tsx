import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LeftMenu from "../index";

describe("<LeftMenu />", () => {
  test("renders", () => {
    const selectedComponent = 'tickets';
    const setSelectedComponent = jest.fn();
    const wrapper = shallow(<LeftMenu selectedComponent={selectedComponent} setSelectedComponent={setSelectedComponent}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});