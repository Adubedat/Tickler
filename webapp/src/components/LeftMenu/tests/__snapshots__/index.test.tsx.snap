// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftMenu /> renders 1`] = `
<styled.div>
  <styled.div
    id="tickets"
    onClick={[Function]}
    selected="tickets"
  >
    <Blueprint3.Icon
      icon="bookmark"
      iconSize={20}
      intent="none"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    />
    <span
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "fontSize": "1.1em",
        }
      }
    >
      Tickets
    </span>
    <Blueprint3.Icon
      icon="chevron-right"
      iconSize={20}
      intent="none"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    />
  </styled.div>
</styled.div>
`;
